src/main/java/org/team1540/robot2024/util/vision/TimestampedVisionPose.java

```diff
@@ -0,0 +1,40 @@
+package org.team1540.robot2024.util.vision;
+
+import edu.wpi.first.math.geometry.Pose2d;
+
+import java.util.Arrays;
+import java.util.Objects;
+
+/**
+ * @param timestampSecs  RIO FPGA timestamp of this vision pose
+ * @param poseMeters     Measured, blue alliance-origin pose from vision
+ * @param seenTagIDs     Array of all tags used to compute this measurement
+ * @param tagPosesMeters Pose in robot-space of all tags used to compute this measurement
+ */
+public record TimestampedVisionPose(
+        double timestampSecs,
+        Pose2d poseMeters,
+        int[] seenTagIDs,
+        Pose2d[] tagPosesMeters) {
+
+    public int getNumTagsSeen() {
+        return seenTagIDs.length;
+    }
+
+    public double getAverageTagDistance() {
```

Comment:
Why?

Reply:
If we wanted to use this data in deciding which camera to trust more or in the pose fusion algorithm having this would be helpful, a lot of this right now is very basic and could probably be made better.